Vic2ToHoI4/Source/HOI4World/HoI4Focus.cpp

```diff
@@ -29,103 +29,109 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.*/
+	registerKeyword("cost", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::singleInt costInt(theStream);
 			cost = costInt.getInt();
 		}
 	);
-	registerKeyword(std::regex("available_if_capitulated"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("available_if_capitulated", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::singleString availableIfCapitulatedString(theStream);
 			availableIfCapitulated = (availableIfCapitulatedString.getString() == "yes");
 		}
 	);
-	registerKeyword(std::regex("available"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("available", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::stringOfObject availableString(theStream);
 			available = availableString.getString();
 		}
 	);
-	registerKeyword(std::regex("cancel_if_invalid"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("cancel_if_invalid", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::singleString cancelIfInvalidString(theStream);
 			cancelIfInvalid = cancelIfInvalidString.getString();
 		}
 	);
-	registerKeyword(std::regex("continue_if_invalid"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("continue_if_invalid", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::singleString continueIfInvalidString(theStream);
 			continueIfInvalid = continueIfInvalidString.getString();
 		}
 	);
-	registerKeyword(std::regex("complete_tooltip"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("select_effect", [this](const std::string& unused, std::istream& theStream)
+		{
+			commonItems::stringOfObject selectEffectString(theStream);
+			selectEffect = selectEffectString.getString();
+		}
+	);
+	registerKeyword("complete_tooltip", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::stringOfObject completeTooltipString(theStream);
 			completeTooltip = completeTooltipString.getString();
 		}
 	);
-	registerKeyword(std::regex("completion_reward"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("completion_reward", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::stringOfObject completionRewardString(theStream);
 			completionReward = completionRewardString.getString();
 		}
 	);
-	registerKeyword(std::regex("ai_will_do"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("ai_will_do", [this](const std::string& unused, std::istream& theStream)
 		{
 			commonItems::stringOfObject aiWillDoString(theStream);
 			aiWillDo = aiWillDoString.getString();
 		}
 	);
-	registerKeyword(std::regex("prerequisite"), [this](const std::string& unused, std::istream& theStream)
+	registerKeyword("prerequisite", [this](const std::string& unused, std::istream& theStream)
```

Comment:
After parsestream, in all parsers you edit or encounter, please put clearRegisteredKeywords() to free up memory. Otherwise - and this happened a lot in eu4tovic2, mem usage would explode.

Before parseStream it's /really/ advised to put a catchall in case someone makes a typo or something changes - otherwise parser would scream "unrecognized token" and die:
registerRegex("[a-zA-Z0-9\\_.:]+", commonItems::ignoreItem);